packages/styles/settings-tools/_s.text-input.scss

```diff
@@ -0,0 +1,76 @@
+$input-sizes: (
+  's': (
+    'font-size': '04',
+    'line-height': 'l',
+    'padding-b': $mu025,
+    'padding-h': $mu050,
+    'padding-t': $mu050,
+  ),
+  'm': (
+    'font-size': '05',
+    'line-height': 'l',
+    'padding-b': $mu075,
+    'padding-h': $mu075,
+    'padding-t': $mu075,
+  ),
+);
+
+$input-icon-sizes: (
+  's': (
+    'left': $mu050,
+    'height': $mu150,
+    'width': $mu150,
+    'padding-l': $mu250,
+  ),
+  'm': (
+    'left': $mu075,
+    'height': $mu150,
+    'width': $mu150,
+    'padding-l': $mu300,
+  ),
+);
+
+$input-default-size: map-get($input-sizes, 'm');
+$input-icon-default-size: map-get($input-icon-sizes, 'm');
+
+@mixin set-input-size($sizeProps) {
+  $border-width: get-border('s');
+  $padding-b: map-get($sizeProps, 'padding-b');
+  $padding-h: map-get($sizeProps, 'padding-h');
+  $padding-t: map-get($sizeProps, 'padding-t');
+
+  @include set-font-scale(
+    map-get($sizeProps, 'font-size'),
+    map-get($sizeProps, 'line-height')
+  );
+
+  // remove border height and center baseline optically
+  padding:
```

Comment:
design : the text does not appear to be optically centered in the small variation, you may need another calculation